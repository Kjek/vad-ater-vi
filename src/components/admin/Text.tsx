import { useCallback, type HTMLAttributes } from 'react';

interface TextProps extends HTMLAttributes<HTMLElement> {
  className?: string;
  variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
  placeholder?: string;
}

const Text = (props: TextProps) => {
  const { className, variant, children } = props;
  const Tag = variant ?? 'span';
  const classes = useCallback(() => {
    switch (variant) {
      case 'h1':
        return 'text-3xl font-bold text-gray-800 dark:text-gray-300';
      case 'h2':
        return 'pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0';
      case 'h3':
        return 'text-2xl font-bold text-gray-800 dark:text-gray-300';
      case 'h4':
        return 'font-bold text-gray-800 dark:text-gray-300';
      case 'h5':
        return 'text-gray-500 dark:text-gray-300';
      case 'h6':
      case 'p':
      case 'span':
      default:
        return 'text-lg text-gray-500 dark:text-gray-300';
    }
  }, [variant])();

  return (
    <>
      <Tag
        {...props}
        title={props.title}
        className={`${className ?? ''} ${classes}`}
      >
        {children}
      </Tag>
    </>
  );
};

export default Text;
