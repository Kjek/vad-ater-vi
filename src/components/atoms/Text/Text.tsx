import { cn } from '@util/cn';
import { useCallback, type HTMLAttributes } from 'react';

interface TextProps extends HTMLAttributes<HTMLElement> {
  className?: string;
  variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
  type?: 'normal' | 'info' | 'error';
  placeholder?: string;
}

const Text = (props: TextProps) => {
  const { className, variant, children, type } = props;
  const Tag = variant ?? 'span';
  const tagClasses = useCallback(() => {
    switch (variant) {
      case 'h1':
        return 'text-3xl font-bold text-gray-800 dark:text-gray-300';
      case 'h2':
        return 'pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0';
      case 'h3':
        return 'text-2xl font-bold text-gray-800 dark:text-gray-300';
      case 'h4':
        return 'font-bold text-gray-800 dark:text-gray-300';
      case 'h5':
        return 'text-gray-500 dark:text-gray-300';
      case 'h6':
      case 'p':
      case 'span':
      default:
        return 'text-lg text-gray-500 dark:text-gray-300';
    }
  }, [variant])();

  const typeClasses = useCallback(() => {
    switch (type) {
      case 'error':
        return 'text-red-700 dark:text-red-500';
      case 'info':
      case 'normal':
        return 'text-gray-500 dark:text-gray-300';
      default:
        return '';
    }
  }, [type])();

  return (
    <>
      <Tag
        {...props}
        title={props.title}
        className={cn(tagClasses, typeClasses, className)}
      >
        {children}
      </Tag>
    </>
  );
};

export default Text;
