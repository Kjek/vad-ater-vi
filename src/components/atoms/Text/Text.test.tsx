import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import Text from './Text';

describe(Text, () => {
  it('renders correctly', () => {
    const { getByTestId } = render(<Text data-testid={'text'} />);
    const textComponent = getByTestId('text');
    expect(textComponent).toBeInTheDocument();
    expect(textComponent).toBeVisible();
  });

  it('displays correctly', () => {
    const { getByTestId } = render(<Text data-testid={'text'}>Test</Text>);
    const textComponent = getByTestId('text');
    expect(textComponent).toBeInTheDocument();
    expect(textComponent).toBeVisible();
    expect(textComponent).toHaveTextContent('Test');
  });

  it('render correct tag', () => {
    const { getByTestId } = render(
      <Text data-testid={'text'} variant='h1'>
        Test
      </Text>
    );
    const textComponent = getByTestId('text');
    expect(textComponent).toBeInTheDocument();
    expect(textComponent).toBeVisible();
    expect(textComponent.tagName).toEqual('H1');
  });
});
