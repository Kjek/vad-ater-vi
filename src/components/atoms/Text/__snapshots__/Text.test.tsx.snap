// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-3xl font-bold text-gray-800 dark:text-gray-300"
    >
      Test
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
  </body>,
  "container": <div>
    <h2
      class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
    >
      Test
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
    <div>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      class="text-2xl font-bold text-gray-800 dark:text-gray-300"
    >
      Test
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
    <div>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h3>
    </div>
    <div>
      <h4
        class="font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h4>
    </div>
  </body>,
  "container": <div>
    <h4
      class="font-bold text-gray-800 dark:text-gray-300"
    >
      Test
    </h4>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 5`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
    <div>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h3>
    </div>
    <div>
      <h4
        class="font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h4>
    </div>
    <div>
      <h5
        class="text-gray-500 dark:text-gray-300"
      >
        Test
      </h5>
    </div>
  </body>,
  "container": <div>
    <h5
      class="text-gray-500 dark:text-gray-300"
    >
      Test
    </h5>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 6`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
    <div>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h3>
    </div>
    <div>
      <h4
        class="font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h4>
    </div>
    <div>
      <h5
        class="text-gray-500 dark:text-gray-300"
      >
        Test
      </h5>
    </div>
    <div>
      <h6
        class="text-lg text-gray-500 dark:text-gray-300"
      >
        Test
      </h6>
    </div>
  </body>,
  "container": <div>
    <h6
      class="text-lg text-gray-500 dark:text-gray-300"
    >
      Test
    </h6>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 7`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
    <div>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h3>
    </div>
    <div>
      <h4
        class="font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h4>
    </div>
    <div>
      <h5
        class="text-gray-500 dark:text-gray-300"
      >
        Test
      </h5>
    </div>
    <div>
      <h6
        class="text-lg text-gray-500 dark:text-gray-300"
      >
        Test
      </h6>
    </div>
    <div>
      <p
        class="text-lg text-gray-500 dark:text-gray-300"
      >
        Test
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="text-lg text-gray-500 dark:text-gray-300"
    >
      Test
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text > matches snapshot 8`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h1>
    </div>
    <div>
      <h2
        class="pb-4 text-center text-3xl font-bold text-gray-800 dark:text-gray-300 sm:w-1/3 sm:grow sm:pb-0"
      >
        Test
      </h2>
    </div>
    <div>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h3>
    </div>
    <div>
      <h4
        class="font-bold text-gray-800 dark:text-gray-300"
      >
        Test
      </h4>
    </div>
    <div>
      <h5
        class="text-gray-500 dark:text-gray-300"
      >
        Test
      </h5>
    </div>
    <div>
      <h6
        class="text-lg text-gray-500 dark:text-gray-300"
      >
        Test
      </h6>
    </div>
    <div>
      <p
        class="text-lg text-gray-500 dark:text-gray-300"
      >
        Test
      </p>
    </div>
    <div>
      <span
        class="text-lg text-gray-500 dark:text-gray-300"
      >
        Test
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="text-lg text-gray-500 dark:text-gray-300"
    >
      Test
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
